src/IO/H5/VolumeData.cpp

```diff
@@ -154,6 +162,308 @@ void append_element_extents_and_connectivity(
+  std::vector<std::pair<size_t, std::array<double, SpatialDim>>>
+      connectivity_of_keys;
+
+  std::array<double, SpatialDim> point_one = {};
+  std::array<double, SpatialDim> point_two = {};
+  std::array<double, SpatialDim> point_three = {};
+  std::array<double, SpatialDim> point_four = {};
+  std::array<double, SpatialDim> point_five = {};
+  std::array<double, SpatialDim> point_six = {};
+  std::array<double, SpatialDim> point_seven = {};
+  std::array<double, SpatialDim> point_eight = {};
+
+  for (size_t i = 0; i < sorted_x.size() - 1; ++i) {
+    point_one[0] = sorted_x[i];
+    point_two[0] = sorted_x[i + 1];
+    if (SpatialDim > 1) {
+      point_three[0] = sorted_x[i + 1];
+      point_four[0] = sorted_x[i];
+      for (size_t j = 0; j < sorted_y.size() - 1; ++j) {
+        point_one[1] = sorted_y[j];
+        point_two[1] = sorted_y[j];
+        point_three[1] = sorted_y[j + 1];
+        point_four[1] = sorted_y[j + 1];
+        if (SpatialDim == 3) {
+          point_five[0] = sorted_x[i];
+          point_six[0] = sorted_x[i + 1];
+          point_seven[0] = sorted_x[i + 1];
+          point_eight[0] = sorted_x[i];
+          point_five[1] = sorted_y[j];
+          point_six[1] = sorted_y[j];
+          point_seven[1] = sorted_y[j + 1];
+          point_eight[1] = sorted_y[j + 1];
+          for (size_t k = 0; k < sorted_z.size() - 1; ++k) {
+            point_one[2] = sorted_z[k];
+            point_two[2] = sorted_z[k];
+            point_three[2] = sorted_z[k];
+            point_four[2] = sorted_z[k];
+            point_five[2] = sorted_z[k + 1];
+            point_six[2] = sorted_z[k + 1];
+            point_seven[2] = sorted_z[k + 1];
+            point_eight[2] = sorted_z[k + 1];
+
+            connectivity_of_keys.insert(
+                connectivity_of_keys.end(),
+                {std::make_pair(block_number, point_one),
+                 std::make_pair(block_number, point_two),
+                 std::make_pair(block_number, point_three),
+                 std::make_pair(block_number, point_four),
+                 std::make_pair(block_number, point_five),
+                 std::make_pair(block_number, point_six),
+                 std::make_pair(block_number, point_seven),
+                 std::make_pair(block_number, point_eight)});
+          }
+        } else {
+          connectivity_of_keys.insert(
+              connectivity_of_keys.end(),
+              {std::make_pair(block_number, point_one),
+               std::make_pair(block_number, point_two),
+               std::make_pair(block_number, point_three),
+               std::make_pair(block_number, point_four)});
+        }
+      }
+    } else {
+      connectivity_of_keys.insert(connectivity_of_keys.end(),
+                                  {std::make_pair(block_number, point_one),
+                                   std::make_pair(block_number, point_two)});
+    }
+  }
+  return connectivity_of_keys;
+}
+
+// Generates the new connectivity
```

Comment:
I think it would be good to document why you can't use `compute_cells` in `Connectivity.?pp` for this. If I've understood correctly it's just that the other one assumes a very specific grid shape.

Reply:
I think I understand what's happening with the `compute_cells`, as it seems like it just does something a little different than is necessary. It looks like it generates the extents in addition to the "cells". 

I think I'm a bit confused on what the "cells" are. Are they the quadrilaterals in 2D and the hexahedrons in 3D? That's to say, are they the smallest fundamental piece in the domain, made up of the minimum number of connections between grid points?

Reply:
ping